pkg/kubelet/cm/container_manager.go

```diff
@@ -122,6 +122,10 @@ type ContainerManager interface {
 	// UnrepareResources unprepares pod resources
 	UnprepareResources(*v1.Pod) error
 
+	// PodCouldHaveUnpreparedResources returns true if the pod with the given UID could still have
+	// unprepared resources.
+	PodCouldHaveUnpreparedResources(*v1.Pod) bool
```

Comment:
This name isn't really accurate for what it is doing under the hood. Resources are only unprepared for the last pod referencing them. So there will be many pods that pass this check (i.e. indicate that they don't have unpepared resources) even though the resources they had claim to are still unprepared (and that is OK).

I don't like this name necessarily, but the semantics we actually want to encode here are `CanPodBeDeletedWithPossibleUnpreparedResources()`. If this is the last pod referenced in any `claimInfo` from the cache, then this returns true, otherwise it returns false.

Reply:
To be clear, if we can't come up with anything better, I think `CanPodBeDeletedWithPossibleUnpreparedResources()` is probably a fine name for now. But I don't think we should use `PodCouldHaveUnpreparedResources()` as it is misleading.

Reply:
I like PodCouldHaveUnpreparedResources better as at the time of calling there could be multiple pods referencing the same claim. At that time we don't know which of them will be the last one, i.e. any of them could have unprepared resources. So the name sounds reasonable to me.

Another point is that PodCouldHaveUnpreparedResources and PodCouldHaveRunningContainers are semantically similar.

CanPodBeDeletedWithPossibleUnpreparedResources suggests that we can choose one pod from multiple existing pods referencing the same claim, which is not possible at the time of calling that function. We can only understand that the claim is still referenced by at least one pod, but we can't conclude which pods can and which can't be deleted if they have running containers referencing the claim.

Reply:
When I read PodCouldHaveUnpreparedResources(pod) I would expect this to return true for ‚pod‘ all the way up until the resources that that pod had a claim to get unprepared. That is not what we want though. We want it to return false for any pod that is not the last to reference a claim, and true for the last pod up until the resources it has a claim to are actually unprepared.

Reply:
as discussed on Slack, we'll use PodMightNeedToUnprepareResources

Reply:
done
tested:
Ran 18 of 6811 Specs in 269.527 seconds
SUCCESS! -- 18 Passed | 0 Failed | 0 Pending | 6793 Skipped